src/client/Microsoft.Identity.Client/Platforms/Features/RuntimeBroker/WamAdapters.cs

```diff
@@ -140,10 +140,11 @@ private enum ResponseStatus
         /// <param name="logger"></param>
         public static NativeInterop.AuthParameters GetCommonAuthParameters(
             AuthenticationRequestParameters authenticationRequestParameters, 
-            WindowsBrokerOptions brokerOptions,
+            BrokerOptions brokerOptions,
             ILoggerAdapter logger)
         {
-            logger.Verbose(() => "[WamBroker] Validating Common Auth Parameters.");
+            logger.Verbose(() => "[RuntimeBroker] Validating Common Auth Parameters.");
+            brokerOptions?.Validate();
```

Comment:
I would move this validation to RuntimeBroker constructor. This way we validate once when PCA is build and RuntimeBroker is instantiated. As-is the validation is postponed until any of the AcquireToken methods are called.

Also I think `brokerOptions` is never null.

Reply:
It is made sure that broker options will be valid. So removed validate method.